src/hdusd/bl_nodes/nodes/shader.py

```diff
@@ -0,0 +1,178 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#********************************************************************
+import math
+
+from ..node_parser import NodeParser
+
+
+class ShaderNodeBsdfPrincipled(NodeParser):
+    def export(self):
+        def enabled(val):
+            if val is None:
+                return False
+
+            if isinstance(val.data, float) and math.isclose(val.data, 0.0):
+                return False
+
+            if isinstance(val.data, tuple) and \
+               math.isclose(val.data[0], 0.0) and \
+               math.isclose(val.data[1], 0.0) and \
+               math.isclose(val.data[2], 0.0):
+                return False
+
+            return True
+
+        # GETTING REQUIRED INPUTS
+        # Note: if some inputs are not needed they won't be taken
+
+        base_color = self.get_input_value('Base Color')
+
+        subsurface = self.get_input_value('Subsurface')
+        subsurface_radius = None
+        subsurface_color = None
+        if enabled(subsurface):
+            subsurface_radius = self.get_input_value('Subsurface Radius')
+            subsurface_color = self.get_input_value('Subsurface Color')
+
+        metallic = self.get_input_value('Metallic')
+        specular = self.get_input_value('Specular')
+        specular_tint = self.get_input_value('Specular Tint')
+        roughness = self.get_input_value('Roughness')
+
+        anisotropic = None
+        anisotropic_rotation = None
+        if enabled(metallic):
+            # TODO: use Specular Tint input
+            anisotropic = self.get_input_value('Anisotropic')
+            if enabled(anisotropic):
+                anisotropic_rotation = self.get_input_value('Anisotropic Rotation')
+                # anisotropic_rotation = 0.5 - (anisotropic_rotation % 1.0)
+
+        sheen = self.get_input_value('Sheen')
+        sheen_tint = None
+        if enabled(sheen):
+            sheen_tint = self.get_input_value('Sheen Tint')
+
+        clearcoat = self.get_input_value('Clearcoat')
+        clearcoat_roughness = None
+        if enabled(clearcoat):
+            clearcoat_roughness = self.get_input_value('Clearcoat Roughness')
+
+        ior = self.get_input_value('IOR')
+
+        transmission = self.get_input_value('Transmission')
+        transmission_roughness = None
+        if enabled(transmission):
+            transmission_roughness = self.get_input_value('Transmission Roughness')
+
+        emission = self.get_input_value('Emission')
+        emission_strength = self.get_input_value('Emission Strength')
+
+        alpha = self.get_input_value('Alpha')
+        # transparency = 1.0 - alpha
+
+        normal = self.get_input_link('Normal')
+        clearcoat_normal = self.get_input_link('Clearcoat Normal')
+        tangent = self.get_input_link('Tangent')
+
+        # CREATING STANDARD SURFACE
+        result = self.create_node('standard_surface', 'surfaceshader', {
```

Comment:
Do we need / want to use Standard Surface or USD Preview Surface

Reply:
USD Preview Surface were not added to MX node tree and I've just added USD Preview Surface nodes. I think here we can use both Standard Surface or USD Preview Surface, but Standard Surface has more options, therefore I prefer to use it here.